### PR TITLE
Passphrase Support for e2e backups

### PR DESCRIPTION
**Pull request by dbkr**

### CODE REVIEW

**Comment by dbkr** [comment]
https://github.com/vector-im/riot-meta/issues/227

File: src/components/views/dialogs/keybackup/CreateKeyBackupDialog.js [hunk 1]
```diff
@@ -98,61 +101,215 @@ export default React.createClass({
...
+        this._keyBackupInfo = await MatrixClientPeg.get().prepareKeyBackupVersion(this.state.passPhrase);
+        this.setState({
+            copied: false,
+            phase: PHASE_SHOWKEY,
+        });
+    },
+
+    _onPassPhraseConfirmKeyPress: function(e) {
+        if (e.key === 'Enter') {
+            this._onPassPhraseConfirmNextClick();
```
Comment (turt2live) [inline src/components/views/dialogs/keybackup/CreateKeyBackupDialog.js]: all the keydown handlers look to give the user a way to submit even when the "next" button is disabled. Should probably have a condition either here or in the click handler to check for that condition.

File: src/components/views/dialogs/keybackup/CreateKeyBackupDialog.js [hunk 2]
```diff
@@ -98,61 +101,215 @@ export default React.createClass({
...
+                    </AccessibleButton>
+                </div>
+            </div>;
+        }
+        const DialogButtons = sdk.getComponent('views.elements.DialogButtons');
+        return <div>
+            <p>
+                Type in your Recovery Passphrase to confirm you remember it.
+                If it helps, add it to your password manager or store it
+                somewhere safe.
```
Comment (turt2live) [inline src/components/views/dialogs/keybackup/CreateKeyBackupDialog.js]: translations

File: src/components/views/dialogs/keybackup/CreateKeyBackupDialog.js [hunk 3]
```diff
@@ -176,6 +333,39 @@ export default React.createClass({
         </div>;
     },
 
+    _renderPhaseOptOutConfirm: function() {
+        const DialogButtons = sdk.getComponent('views.elements.DialogButtons');
+        return <div>
+            Without setting up Secure Message Recovery, you won't be able to restore your
+            encrypted message history if you log out or use another device.
```
Comment (turt2live) [inline src/components/views/dialogs/keybackup/CreateKeyBackupDialog.js]: transltions

File: src/components/views/dialogs/keybackup/CreateKeyBackupDialog.js (hunk 1 above)
Comment (dbkr) [reply src/components/views/dialogs/keybackup/CreateKeyBackupDialog.js]: ooh, good catch

File: src/components/views/dialogs/keybackup/CreateKeyBackupDialog.js (hunk 3 above)
Comment (dbkr) [reply src/components/views/dialogs/keybackup/CreateKeyBackupDialog.js]: ooh, thanks

**Comment by dbkr** [comment]
right, that should be better

**Pull request merged**